stac_generator/point_stac_generator.py

```diff
@@ -0,0 +1,353 @@
+
+    Args:
+        df (FrameT): input dataframe containing X_coord and Y_coord columns
+        X_coord (str): name of X column
+        Y_coord (str): name of Y column
+
+    Returns:
+        BBoxT: Bounding box coordinate
+    """
+    min_X, max_X = float(df[X_coord].min()), float(df[X_coord].max())
+    min_Y, max_Y = float(df[Y_coord].min()), float(df[Y_coord].max())
+    return [min_X, min_Y, max_X, max_Y]
+
+
+def calculate_temporal_extent(
+    df: FrameT | None = None,
+    time_col: str | None = None,
+    datetime: DateTimeT | None = None,
+    start_datetime: DateTimeT | None = None,
+    end_datetime: DateTimeT | None = None,
+) -> TimeExtentT:
+    """Get temporal extent based on STAC specification. Use start_datetime and end_datetime if provided.
+    Otherwise use datetime if provided. If the dataframe and time col is provided, will obtain the
+    start datetime, end date time from the dataframe. Otherwise get current UTC time.
+
+    Args:
+        df (FrameT | None, optional): provided dataframe. Defaults to None.
+        time_col (str | None, optional): name of time column. Defaults to None.
+        datetime (DateTimeT | None, optional): datetime. Defaults to None.
+        start_datetime (DateTimeT | None, optional): start_datetime. Defaults to None.
+        end_datetime (DateTimeT | None, optional): end_datetime. Defaults to None.
+
+    Returns:
+        TimeExtentT: start datetime, end_datetime
+    """
+    if start_datetime and end_datetime:
+        return start_datetime, end_datetime
+    if datetime:
+        return None, datetime
+    if df is not None and isinstance(time_col, str):
+        if time_col not in df.columns:
+            raise KeyError(f"Cannot find time_col: {time_col} in given dataframe")
+        if not isinstance(df[time_col].dtype, DateTimeT):
+            raise ValueError(
+                f"Dtype of time_col: {time_col} must be of datetime type: {df[time_col].dtype}"
+            )
+        min_T, max_T = df[time_col].min(), df[time_col].max()
+        return (min_T, max_T)
+    return None, pydatetime.datetime.now(pydatetime.UTC)
+
+
+def calculate_geometry(
+    df: FrameT,
+    X_coord: str = "X",
+    Y_coord: str = "Y",
+) -> GeometryObj:
+    coords = np.unique(df.loc[:, [X_coord, Y_coord]].values, axis=0)
+    coord_type: Literal["Point", "MultiPoint"] = "Point" if len(coords) == 1 else "MultiPoint"
+    return GeometryObj(type=coord_type, coordinates=coords)
+
+
+def read_csv(
+    src_path: str,
+    X_coord: str,
+    Y_coord: str,
+    T_coord: str | None = None,
+    date_format: str = "ISO8601",
+    bands: Sequence[str] | Sequence[PointBandInfo] | None = None,
+    item_group: Sequence[str] | None = None,
+) -> FrameT:
+    parse_dates = [T_coord] if isinstance(T_coord, str) else False
+    usecols = None
+    # If band info is provided, only read in the required columns + the X and Y coordinates
+    if bands:
```

Comment:
My gut feeling is to refer to this variable `bands` with a different name to avoid confusing it with raster data. Maybe `data_variables`? @dhobern do you have any thoughts? I can explain it better over a call if necessary.

Reply:
Personally, I think of the data_variables as bands when we start adding them as layers onto a map. They are just as much "bands" as NDVI is. But I don't have a strong opinion, particularly if it is just a local name for this method as it seems to be here.

Reply:
I see bands is a parameter to the constructor, so not just the local name here, but I remain easy either way.